opensrp-opd/src/main/java/org/smartregister/opd/activity/BaseOpdRegisterActivity.java

```diff
@@ -112,39 +113,43 @@ public void startFormActivity(String formName, String entityId, String metaData,
             String locationId = OpdUtils.context().allSharedPreferences().getPreference(AllConstants.CURRENT_LOCATION_ID);
             presenter().startForm(formName, entityId, metaData, locationId, injectedFieldValues, entityTable);
         } else {
-
             displayToast(getString(R.string.error_unable_to_start_form));
         }
     }
 
     @Override
     public void startFormActivity(@NonNull JSONObject jsonForm, @Nullable HashMap<String, String> parcelableData) {
-
-        Intent intent = new Intent(this, OpdLibrary.getInstance().getOpdConfiguration().getOpdMetadata().getOpdFormActivity());
-        intent.putExtra(OpdConstants.JSON_FORM_EXTRA.JSON, jsonForm.toString());
-        Form form = new Form();
-        form.setWizard(true);
-        form.setName(OpdConstants.EventType.DIAGNOSIS_AND_TREAT);
-
-        String encounterType = jsonForm.optString(OpdJsonFormUtils.ENCOUNTER_TYPE);
-        if (encounterType.equals(OpdConstants.EventType.CHECK_IN)) {
+        OpdMetadata opdMetadata = OpdLibrary.getInstance().getOpdConfiguration().getOpdMetadata();
+        if (opdMetadata != null) {
+            Intent intent = new Intent(this, opdMetadata.getOpdFormActivity());
+            intent.putExtra(OpdConstants.JSON_FORM_EXTRA.JSON, jsonForm.toString());
+            Form form = new Form();
             form.setWizard(false);
             form.setName("");
-        }
 
-        form.setHideSaveLabel(true);
-        form.setPreviousLabel("");
-        form.setNextLabel("");
-        form.setHideNextButton(false);
-        form.setHidePreviousButton(false);
-        intent.putExtra(JsonFormConstants.JSON_FORM_KEY.FORM, form);
+            String encounterType = jsonForm.optString(OpdJsonFormUtils.ENCOUNTER_TYPE);
 
-        if (parcelableData != null) {
-            for (String intentKey : parcelableData.keySet()) {
-                intent.putExtra(intentKey, parcelableData.get(intentKey));
+            if (encounterType.equals(OpdConstants.EventType.DIAGNOSIS_AND_TREAT)) {
+                form.setName(OpdConstants.EventType.DIAGNOSIS_AND_TREAT);
+                form.setWizard(true);
             }
-        }
 
-        startActivityForResult(intent, OpdJsonFormUtils.REQUEST_CODE_GET_JSON);
+            form.setHideSaveLabel(true);
+            form.setPreviousLabel("");
+            form.setNextLabel("");
+            form.setHideNextButton(false);
+            form.setHidePreviousButton(false);
+
+            intent.putExtra(JsonFormConstants.JSON_FORM_KEY.FORM, form);
+            if (parcelableData != null) {
+                for (String intentKey : parcelableData.keySet()) {
+                    intent.putExtra(intentKey, parcelableData.get(intentKey));
+                }
+            }
+
+            startActivityForResult(intent, OpdJsonFormUtils.REQUEST_CODE_GET_JSON);
+        } else {
+            Timber.e(new Exception(), "FormActivity cannot be started because OpdMetadata is NULL");
```

Comment:
I think we do, I need the stacktrace and exception location otherwise, only the message will be printed and I have to search for the message throughout the codebase